### PR TITLE
[CI] Install igc dev in devigc docker image

### PR DESCRIPTION
**Pull request by jsji**
- Add flag in install_driver to install igc dev driver
- Add options to build devigc containers

Ref: https://github.com/intel/llvm/pull/11552

### CODE REVIEW

File: .github/workflows/sycl-containers.yaml [hunk 1]
```diff
@@ -93,6 +93,33 @@ jobs:
...
+          push: ${{ github.event_name != 'pull_request' }}
+          file: ubuntu2204_intel_drivers
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          tags: |
+            ghcr.io/${{ github.repository }}/ubuntu2204_intel_drivers:devigc-${{ github.sha }}
+            ghcr.io/${{ github.repository }}/ubuntu2204_intel_drivers:devigc
+          build-args: |
+            use_latest=false
+            use_igc_dev=true
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-containers.yaml:122]: Would it be possible to refactor this code to use a `matrix`? See https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl-linux-matrix-e2e-on-nightly.yml for an example.

Comment (jsji) [reply .github/workflows/sycl-containers.yaml:122]: Sure, can do the factor in a follow up PR.

Comment (aelovikov-intel) [reply .github/workflows/sycl-containers.yaml:122]: Can we do that in the reverse order? Refactor first, then add new code.

Comment (jsji) [reply .github/workflows/sycl-containers.yaml:122]: Fair enough.

**Comment by jsji** [comment]
Rebased to latest after refactoring.

File: devops/containers/ubuntu2204_intel_drivers.Dockerfile [hunk 2]
```diff
@@ -6,8 +6,9 @@ FROM $base_image:$base_tag
 ENV DEBIAN_FRONTEND=noninteractive
 
 ARG use_latest=true
+ARG use_igc_dev=false
 
-RUN apt update && apt install -yqq wget
+RUN apt update && apt install -yqq wget libllvm14
```
Comment (aelovikov-intel) [inline devops/containers/ubuntu2204_intel_drivers.Dockerfile:11]: I'm not sure we want LLVM installed in the image unconditionally.

**Review by aelovikov-intel — APPROVED** [review]

File: devops/containers/ubuntu2204_intel_drivers.Dockerfile (hunk 2 above)
Comment (jsji) [reply devops/containers/ubuntu2204_intel_drivers.Dockerfile]: yeah, thought about that too, but this is lib only,  it is not harmful to have it. :)
But sure, let us avoid polluting existing images.
Done.

**Comment by jsji** [comment]
@intel/llvm-gatekeepers  Please help to merge this. Thanks.

File: .github/workflows/sycl-containers.yaml [hunk 3]
```diff
@@ -45,6 +45,12 @@ jobs:
             file: ubuntu2204_intel_drivers
             tags: ubuntu2204_intel_drivers
             build_args: "use_latest=false"
+          - name: Intel Drivers Ubuntu 22.04 Docker image with dev IGC
+            file: ubuntu2204_intel_drivers
+            tags: ubuntu2204_intel_drivers_devigc
+            build_args: |
+                  "use_latest=false"
+                  "use_igc_dev=true"
```
Comment (bader) [inline .github/workflows/sycl-containers.yaml:53]: @jsji, please, document new image in https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/DockerBKMs.md#sycl-containers-overview.

Comment (jsji) [reply .github/workflows/sycl-containers.yaml:53]: Done in https://github.com/intel/llvm/pull/13307

**Pull request merged**